src/spec/items.spec.js

```diff
@@ -115,6 +115,49 @@ describe('items', () => {
     res.body.should.be.an('array');
     res.body.length.should.eq(0);
   });
+  it('should return an item by a non-default key', async () => {
+    let res = await chai.request(server).get('/items/bronco.png?by=imageName');
+    res.should.have.status(200);
+    res.body.should.be.an('object');
+    Object.keys(res.body).length.should.be.greaterThan(0);
+  });
+  it('should return multiple items by a non-default key', async () => {
+    let res = await chai.request(server).get('/items/search/bronco.png?by=imageName');
+    res.should.have.status(200);
+    res.body.should.be.an('array');
+    res.body.length.should.be.greaterThan(0);
+  });
+  it('should return an item by nested keys', async () => {
+    let res = await chai.request(server).get('/items/rare?by=components.drops.rarity');
+    res.should.have.status(200);
+    res.body.should.be.an('object');
+    Object.keys(res.body).length.should.be.greaterThan(0);
+  });
+  it('should give error for unmatchable nested keys', async () => {
+    let res = await chai.request(server).get('/items/bronco?by=components.shoobedowopwah');
+    res.should.have.status(404);
+    res.body.should.be.an('object');
```

Comment:
should be null if it's not found and we're doing single object.   
need a test mirroring this one that uses `/items/search/bronco?...` and verifies that it's an empty array

Reply:
Unmatched single item queries don't return a null body, they return a body with an 'error' string of 'No Result' and a 'code' value of '404'.

Empty array version of this test added to 'should return empty array for unmatchable nested keys' test in commit de72f1c